src/tophu/label.py

```diff
@@ -0,0 +1,287 @@
+        if s.start is None:
+            new_start = None
+        else:
+            # Subtract `margin` from the start index of the slice. We need to be careful
+            # to avoid changing the sign of the index since negative indices in Python
+            # typically wrap around.
+            sign_change = (s.start >= 0) and (s.start < margin)
+            new_start = 0 if sign_change else s.start - margin
+
+        if s.stop is None:
+            new_stop = None
+        else:
+            # Add `margin` to the stop index of the slice. Avoid changing the sign of
+            # the index.
+            sign_change = (s.stop < 0) and (s.stop >= -margin)
+            new_stop = None if sign_change else s.stop + margin
+
+        return slice(new_start, new_stop, s.step)
+
+    return tuple([expand_slice(s) for s in tile])
+
+
+def unique_nonzero_integers(x: ArrayLike, /) -> Set:
+    """
+    Find unique nonzero elements in the input array.
+
+    Parameters
+    ----------
+    x : array_like
+        An array of integers.
+
+    Returns
+    -------
+    s : set
+        The set of unique nonzero values.
+    """
+    return set(np.unique(x)) - {0}
+
+
+def find_connected_labels(
+    label: int,
+    conncomp: NDArray[np.unsignedinteger],
+    *,
+    connectivity: int = 1,
+) -> Set[int]:
+    """
+    Find equivalent labels based on pixel connectivity.
+
+    Determine the set of equivalent labels to the specified label. Two nonzero labels
+    are considered equivalent if the `conncomp` array contains any two connected pixels
+    with those respective labels.
+
+    Zero-valued pixels are ignored.
+
+    Parameters
+    ----------
+    label : int
+        The query label.
+    conncomp : numpy.ndarray
+        A two-dimensional array of positive integer labels associated with connected
+        components. Some connected components may have more than one label. Zero-valued
+        pixels are treated as masked-out (i.e. not part of any connected component).
+    connectivity : int, optional
+        Determines which pixels are considered connected. Valid values are 1 or 2. If
+        `connectivity=1`, only a pixel's 4 orthogonal neighbors are considered
+        connected. If `connectivity=2`, a pixel's 8 orthogonal and diagonal neighbors
+        are considered connected. (default: 1)
+
+    Returns
+    -------
+    equivalent_labels : set of int
+        A set of connected component labels that are equivalent to the specified label.
+    """
+    if connectivity not in {1, 2}:
+        raise ValueError("connectivity must be 1 or 2")
+
+    # Get a mask of pixels in `conncomp` with value equal to `label`.
+    mask = conncomp == label
+
+    # Dilate the masked region to include bordering pixels.
+    struct = ndimage.generate_binary_structure(conncomp.ndim, connectivity)
+    dilated_mask = ndimage.binary_dilation(mask, struct)
+
+    # Get a mask of only pixels bordering the labeled region.
+    edge_mask = dilated_mask ^ mask
```

Comment:
Is this the same as doing 
```python
edge_mask = dilated_mask.copy()
edge_mask[mask] = False
```

Not necessarily saying it's better, just wanted to check since my intuition on bitwise xors is sometimes shaky (and I had to double check that `^` was xor)

Reply:
Ah good suggestion. I agree that your implementation is more intuitive.

Originally, I had this implemented as 

```python
edge_mask = dilated_mask - mask
```

but a numpy error message informed me that subtraction wasn't supported for boolean arrays and suggested that I use xor instead.

I suppose alternatively this could be implemented on a single line as

```python
edge_mask = dilated_mask & ~mask
```

Does that seem sufficiently readable, or do you prefer the two-liner?